### PR TITLE
Adds 64-bit compatibility

### PR DESCRIPTION
**Pull request by cjakeman**
Blueprint https://blueprints.launchpad.net/or/+spec/64bit
Forum thread http://www.elvastower.com/forums/index.php?/topic/33281-64bit-version-4gb-ram-usage/

This uses AnyCPU and runs as 64-bit on 64-bit Windows and as 32-bit on 32-bit Windows.
No Launcher64/Openrails64.exe is included.
No changes to RailDriver are included, so that will have to be a separate PR.

~~I've marked this as DRAFT so it can be tried in Unstable but can't go into Testing until we also have a RailDriver PR.~~
~~Otherwise RailDrivers will stop working on 64-bit systems.~~

~~I've marked this as DRAFT so it can be tried in Unstable but can't go into Testing until the Large Address Aware option has been removed from Menu > Options and Build.cmd~~

### CODE REVIEW

**Review by Sharpe49 — COMMENTED** [review]
That PR will collide with the project files upgrade I was about to push...

File: Source/Contrib/ActivityEditor/AEWizard/AEWizard.csproj [hunk 1]
```diff
@@ -22,6 +22,8 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
[the remark this answers is not in the document]
Comment (twpol) [reply Source/Contrib/ActivityEditor/AEWizard/AEWizard.csproj:26]: This is unfortunately what happens when Git tries to merge VS project files with the recommended `.gitattributes` configuration (which we have in the repo): https://github.com/openrails/openrails/blob/e3e322f19d19cfdf9f11d2ec941339a5941f2ad1/.gitattributes#L42

It's a good behaviour for e.g. new files or dependencies added to a project, but gets confused by items like this which should only appear once (in a section)

We can likely turn it off after your project file upgrade to the new short form, though :)

Comment (Sharpe49) [inline Source/Contrib/ActivityEditor/AEWizard/AEWizard.csproj:26]:
```suggestion
    <PlatformTarget>AnyCPU</PlatformTarget>
```

File: Source/Contrib/ActivityEditor/ActivityEditor/ActivityEditor.csproj [hunk 2]
```diff
@@ -23,6 +23,8 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (Sharpe49) [inline Source/Contrib/ActivityEditor/ActivityEditor/ActivityEditor.csproj:27]:
```suggestion
    <PlatformTarget>AnyCPU</PlatformTarget>
```

File: Source/Contrib/TrackViewer/Drawing/DrawArea.cs [hunk 3]
```diff
@@ -188,10 +188,10 @@ public void Update()
...
         {
             if (drawTrackDB == null) return;
-            float minX = drawTrackDB.MinTileX * 2048 - 1024f;
-            float minZ = drawTrackDB.MinTileZ * 2048 - 1024f;
-            float maxX = drawTrackDB.MaxTileX * 2048 + 1024f;
-            float maxZ = drawTrackDB.MaxTileZ * 2048 + 1024f;
+            double minX = drawTrackDB.MinTileX * WorldLocation.TileSize - 1024f;
+            double minZ = drawTrackDB.MinTileZ * WorldLocation.TileSize - 1024f;
+            double maxX = drawTrackDB.MaxTileX * WorldLocation.TileSize + 1024f;
+            double maxZ = drawTrackDB.MaxTileZ * WorldLocation.TileSize + 1024f;
```
Comment (Sharpe49) [inline Source/Contrib/TrackViewer/Drawing/DrawArea.cs:194]: Why are those calculations switched from float to double?
I doesn't seem related to the switch to 64bit.

File: Source/ORTS.sln [hunk 4]
```diff
@@ -55,110 +55,314 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ContentChecker", "ContentCh
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
 		Debug|Mixed Platforms = Debug|Mixed Platforms
+		Debug|x64 = Debug|x64
+		Release|Any CPU = Release|Any CPU
 		Release|Mixed Platforms = Release|Mixed Platforms
+		Release|x64 = Release|x64
```
Comment (Sharpe49) [inline Source/ORTS.sln:63]: I think the only configuration that should exist is Any CPU. The others can be deleted.

**Review by Sharpe49 — CHANGES_REQUESTED** [review]

File: Source/Contrib/ActivityEditor/AEWizard/AEWizard.csproj (hunk 1 above)
Comment (Sharpe49) [reply Source/Contrib/ActivityEditor/AEWizard/AEWizard.csproj:26]: This suggested change should be applied to all CSPROJ files.

File: Source/Contrib/TrackViewer/Drawing/DrawArea.cs (hunk 3 above)
Comment (cjakeman) [reply Source/Contrib/TrackViewer/Drawing/DrawArea.cs:194]: Good question.

In both the Activity Editor and in RunActivities MapWindow, paths drawn at an angle show distinct wiggles in 64-bit. PerpetualKid (Steffen) found that this change resolved the issue, but it's not clear why.

File: Source/ORTS.sln (hunk 4 above)
Comment (cjakeman) [reply Source/ORTS.sln]: Thanks.

Done in 3b12088

**Review by Sharpe49 — COMMENTED** [review]
You are getting build errors because the csproj files contains "Condition="'$(Configuration)|$(Platform)' == 'Release|x86'"" conditions.
You need to replace x86 by AnyCPU in all csproj files.

File: Source/Contrib/ActivityEditor/ActivityEditor/ActivityEditor.csproj (hunk 2 above)
Comment (Sharpe49) [reply Source/Contrib/ActivityEditor/ActivityEditor/ActivityEditor.csproj]: This change must be applied to all csproj files.

**Comment by cjakeman** [comment]
Hi Cedric, 

> You are getting build errors because the csproj files contains "Condition="'$(Configuration)|$(Platform)' == 'Release|x86'"" conditions. You need to replace x86 by AnyCPU in all csproj files.

Thanks for this - exactly what was needed.
Didn't work at first (lots of bad references) but fiddled with VS a bit and then it behaved itself.

File: Source/RunActivity/RunActivity.csproj [hunk 5]
```diff
@@ -205,7 +205,7 @@
   </ItemGroup>
   <ItemGroup>
     <Content Include="..\3rdPartyLibs\openal-soft\Win32\OpenAL32.dll">
-      <Link>Native\X86\OpenAL32.dll</Link>
+      <Link>Native\AnyCPU\OpenAL32.dll</Link>
```
Comment (perpetualKid) [inline Source/RunActivity/RunActivity.csproj:208]: this seems wrong (AnyCPU). OpenAL32 has 32bit version and a 64bit version, though both are named OpenAL32. They are both copied to output so the corresponding version can be located and loaded at runtime, which is set in Source/RunActivity/Program.cs expecting the dlls in folder x86 for 32bit and x64 for 64bit:
https://github.com/openrails/openrails/blob/8d428b8cb38456442d04374d5a67461f5598074e/Source/RunActivity/Program.cs#L60-L61

with your change the x86 version is copied to a folder AnyCPU, and can't be found at runtime:
http://www.elvastower.com/forums/index.php?/topic/35959-problem-running-or-after-upgrading/

Comment (cjakeman) [reply Source/RunActivity/RunActivity.csproj:208]: Thanks, Steffen.

This PR is now marked as not-for-unstable and I'll test out your change tomorrow before unmarking it.

Comment (cjakeman) [reply Source/RunActivity/RunActivity.csproj]: All done. I think this bad change was the victim of a global search and replace.

File: Build.cmd [hunk 6]
```diff
@@ -109,11 +109,6 @@ REM Run unit tests (9009 means XUnit itself wasn't found, which is an error).
 xunit.console.x86 Program\Tests.dll -nunit xunit.xml
 IF "%ERRORLEVEL%" == "9009" GOTO :error
 
-CALL :copy "Program\RunActivity.exe" "Program\RunActivityLAA.exe" || GOTO :error
-editbin /NOLOGO /LARGEADDRESSAWARE "Program\RunActivityLAA.exe" || GOTO :error
```
Comment (twpol) [inline Build.cmd:113]: I think you can remove the `editbin` check at the start of `Build.cmd` as this is the only use of it

File: Source/Contrib/ActivityEditor/LibAE/LibAE.csproj [hunk 7]
```diff
@@ -13,26 +13,28 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG;WINDOWS;ACTIVITY_EDITOR;JSON_OR_XML</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Contrib/ActivityEditor/LibAE/LibAE.csproj:26]: Still got some duplication from Git's merge here

File: Source/Contrib/ContentManager/ContentManager.csproj [hunk 8]
```diff
@@ -14,26 +14,28 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Contrib/ContentManager/ContentManager.csproj:27]: Duplicated here too

File: Source/Contrib/DataCollector/DataCollector.csproj [hunk 9]
```diff
@@ -13,26 +13,28 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Contrib/DataCollector/DataCollector.csproj:26]: And here

File: Source/Contrib/TrackViewer/TrackViewer.csproj [hunk 10]
```diff
@@ -14,26 +14,28 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG;WINDOWS</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Contrib/TrackViewer/TrackViewer.csproj:27]: And one more

File: Source/Menu/Menu.csproj [hunk 11]
```diff
@@ -14,27 +14,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Menu/Menu.csproj:27]: Another one

File: Source/ORTS.Common/ORTS.Common.csproj [hunk 12]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/ORTS.Common/ORTS.Common.csproj:26]: Another

File: Source/ORTS.Menu/ORTS.Menu.csproj [hunk 13]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/ORTS.Menu/ORTS.Menu.csproj:26]: Another

File: Source/ORTS.Settings/ORTS.Settings.csproj [hunk 14]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/ORTS.Settings/ORTS.Settings.csproj:26]: Another

File: Source/ORTS.Updater/ORTS.Updater.csproj [hunk 15]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/ORTS.Updater/ORTS.Updater.csproj:26]: Another

File: Source/ORTS.sln [hunk 16]
```diff
@@ -55,110 +55,310 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ContentChecker", "ContentCh
...
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|Mixed Platforms.ActiveCfg = Debug|x86
 		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|Mixed Platforms.Build.0 = Debug|x86
+		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{9BE44F64-417D-4AA6-AF4D-2A6A0CBC814D}.Debug|x64.Build.0 = Debug|Any CPU
```
Comment (twpol) [inline Source/ORTS.sln:67]: Not sure what's going on here, this looks like each project is going from just "Mixed Platforms" to having three configurations ("Mixed Platforms", "Any CPU", "x64"), which is surely wrong? We should only have "Any CPU" configurations, right?

File: Source/Orts.Formats.Msts/Orts.Formats.Msts.csproj [hunk 17]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Orts.Formats.Msts/Orts.Formats.Msts.csproj:26]: Another duplicate

File: Source/Orts.Formats.OR/Orts.Formats.OR.csproj [hunk 18]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG;ACTIVITY_EDITOR</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Orts.Formats.OR/Orts.Formats.OR.csproj:26]: Another

File: Source/Orts.Parsers.Msts/Orts.Parsers.Msts.csproj [hunk 19]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Orts.Parsers.Msts/Orts.Parsers.Msts.csproj:26]: Another

File: Source/Orts.Simulation/Orts.Simulation.csproj [hunk 20]
```diff
@@ -15,27 +15,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>TRACE;DEBUG;ACTIVITY_EDITOR</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Orts.Simulation/Orts.Simulation.csproj:28]: Another

File: Source/Tests/Tests.csproj [hunk 21]
```diff
@@ -17,27 +17,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Tests/Tests.csproj:30]: Another

File: Source/Updater/Updater.csproj [hunk 22]
```diff
@@ -13,27 +13,29 @@
...
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>..\..\Program\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <PlatformTarget>x86</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <PlatformTarget>AnyCPU</PlatformTarget>
```
Comment (twpol) [inline Source/Updater/Updater.csproj:26]: And a final one

**Review by twpol — CHANGES_REQUESTED** [review]
Most of the changes look good, but there's something weird about the solution file, and a lot of duplication still in the project files

Both might have to be edited by hand :(

File: Build.cmd (hunk 6 above)
Comment (cjakeman) [reply Build.cmd:113]: Hi James,

Good catch

Chris

File: Source/Contrib/ContentManager/ContentManager.csproj (hunk 8 above)
Comment (cjakeman) [reply Source/Contrib/ContentManager/ContentManager.csproj]: Done

File: Source/Contrib/TrackViewer/TrackViewer.csproj (hunk 10 above)
Comment (cjakeman) [reply Source/Contrib/TrackViewer/TrackViewer.csproj]: Done

File: Source/Contrib/DataCollector/DataCollector.csproj (hunk 9 above)
Comment (cjakeman) [reply Source/Contrib/DataCollector/DataCollector.csproj]: Done

File: Source/ORTS.Common/ORTS.Common.csproj (hunk 12 above)
Comment (cjakeman) [reply Source/ORTS.Common/ORTS.Common.csproj]: Done

File: Source/ORTS.Menu/ORTS.Menu.csproj (hunk 13 above)
Comment (cjakeman) [reply Source/ORTS.Menu/ORTS.Menu.csproj]: Done

File: Source/ORTS.Updater/ORTS.Updater.csproj (hunk 15 above)
Comment (cjakeman) [reply Source/ORTS.Updater/ORTS.Updater.csproj]: Done

File: Source/Orts.Formats.Msts/Orts.Formats.Msts.csproj (hunk 17 above)
Comment (cjakeman) [reply Source/Orts.Formats.Msts/Orts.Formats.Msts.csproj]: Done

File: Source/Orts.Formats.OR/Orts.Formats.OR.csproj (hunk 18 above)
Comment (cjakeman) [reply Source/Orts.Formats.OR/Orts.Formats.OR.csproj]: Done

File: Source/Orts.Simulation/Orts.Simulation.csproj (hunk 20 above)
Comment (cjakeman) [reply Source/Orts.Simulation/Orts.Simulation.csproj]: Done

File: Source/Tests/Tests.csproj (hunk 21 above)
Comment (cjakeman) [reply Source/Tests/Tests.csproj]: Done

Comment (cjakeman) [reply Source/Tests/Tests.csproj]: Done

File: Source/Updater/Updater.csproj (hunk 22 above)
Comment (cjakeman) [reply Source/Updater/Updater.csproj]: Done. Checked and you found them all.

File: Source/ORTS.sln (hunk 16 above)
Comment (cjakeman) [reply Source/ORTS.sln]: Seems to work with only "Any CPU" so expect you are right.

**Review by twpol — APPROVED** [review]
Looks good, thanks!

**Review by Sharpe49 — APPROVED** [review]

**Comment by cjakeman** [comment]
Thanks, Cedric
